### PR TITLE
Search event by keyword (I can't figure out how to delete this...)

### PR DESCRIPTION
**Pull request by mertzjl91**
This allows a search of events by keyword. Only searches the title, description, and location. Doesn't scroll through the price, date or time. Those can be filtered on a later feature.
Message displays when there are no matches.

### CODE REVIEW

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -3,12 +3,15 @@ import {RouterModule, Routes} from '@angular/router';
 import { UserFormComponent } from './user-form/user-form.component';
 import { EventListComponent } from './event-list/event-list.component'
 import { EventFormComponent } from './event-form/event-form.component';
+import { SearchResultsComponent } from './search-results/search-results.component';
+
 
 const routes: Routes = [
   // {path: 'list-user', redirectTo: '/', pathMatch: 'full'},
```
Comment (miketully426) [inline src/app/app-routing.module.ts]: Do we need the commented code?

File: src/app/app-routing.module.ts [hunk 2]
```diff
@@ -3,12 +3,15 @@ import {RouterModule, Routes} from '@angular/router';
...
   { path: 'events', component: EventListComponent },
   {path: 'addevent', component: EventFormComponent},
```
Comment (miketully426) [inline src/app/app-routing.module.ts]: For consistency, can we add a space at the beginning and end of code within the curly braces?

**Review by miketully426 — DISMISSED** [review]

**Comment by miketully426** [comment]
Bravo on adding testing!

File: src/app/app.module.ts [hunk 3]
```diff
@@ -9,13 +9,17 @@ import { UserService } from './service/user-service.service'
...
+import { SearchResultsComponent } from './search-results/search-results.component';
 
 @NgModule({
   declarations: [
     AppComponent,
     EventListComponent,
     EventFormComponent,
+    SearchResultsComponent,
     UserFormComponent,
+
```
Comment (lfallert) [inline src/app/app.module.ts]: do we need these empty lines here?

**Review by lfallert — COMMENTED** [review]
do we need that extra line of code in our component declarations on app.module.ts?
I run an error when I search for an event (TypeError: Cannot read properties of undefined (reading 'length')). But the code works as intended otherwise.

File: src/app/app.module.ts (hunk 3 above)
Comment (mertzjl91) [reply src/app/app.module.ts]: Removed!

**Comment by mertzjl91** [comment]
> do we need that extra line of code in our component declarations on app.module.ts? I run an error when I search for an event (TypeError: Cannot read properties of undefined (reading 'length')). But the code works as intended otherwise.

@lfallert fixed this! I had forgotten this happened, but I worked out a way to fix it and get the error message back up and running. It was an issue with reading the length of the array before it was actually declared in the search function. So I declared it specifically as an empty array in the class and then added a bit to my if statement on the error message in the html to account for the array now being at length 0 when the page loads!

**Review by lfallert — APPROVED** [review]
great job! search results by keyword is working as intended! also lets you search for symbols, numbers, and is case insensitive! Awesome!

**Review by CNichole726 — APPROVED** [review]
Looks good

**Pull request closed without merging**